### PR TITLE
Utilities to prune Builds

### PR DESCRIPTION
**Pull request by derekwaynecarr**
The following adds utility code to support prune build operations.

A follow-on PR will tie it into an ```osadm``` command.

@csrwng @bparees @smarterclayton - this was tedious, I imagine reviewing it will be tedious as well.

### CODE REVIEW

File: pkg/build/prune/data.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+// BuildByBuildConfigIndexFunc indexes Build items by their associated BuildConfig, if none, index with key "orphan"
+func BuildByBuildConfigIndexFunc(obj interface{}) (string, error) {
+	build, ok := obj.(*buildapi.Build)
+	if !ok {
+		return "", fmt.Errorf("Not a Build: %v", build)
```
Comment (mfojtik) [inline pkg/build/prune/data.go]: error messages are in lowercase afaik.

**Comment by derekwaynecarr** [comment]
If I can get a general lgtm on pattern, I would feel better since I have the same massive work to go do for deployments and planned to follow same structure.

**Comment by derekwaynecarr** [comment]
I am going to re-work filter behavior more after seeing more of what deployments will need.

**Comment by derekwaynecarr** [comment]
I think this is ready for a final review now.  When I tackle deployments, I will follow the same pattern.

**Comment by csrwng** [comment]
Other than the issue with error messages (there's a few) that @mfojtik mentioned, LGTM

**Comment by derekwaynecarr** [comment]
Fixed error message, [merge]

**Pull request merged**